### PR TITLE
Moved modCount to file where it is used

### PR DESCRIPTION
**Pull request by dima-avdeev-jb**
`modCount` it is AbstractMutableList property with `protected` modifier.
 This workaround needs to properly compile on Native and JVM platforms.
Don't tested it on JS yet.

### CODE REVIEW

File: compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/external/kotlinx/collections/immutable/internal/commonFunctions.kt [hunk 1]
```diff
@@ -1,11 +0,0 @@
-/*
```
Comment (eymar) [inline compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/external/kotlinx/collections/immutable/internal/commonFunctions.kt]: What was the compile error?
Do I understand the workaround correctly, that it just moved the function from one package to another?

Also, you mentioned it didn't compile properly for JVM too, so android had this issue too?

File: compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/external/kotlinx/collections/immutable/internal/commonFunctions.kt [hunk 2]
```diff
@@ -1,11 +0,0 @@
...
- * Copyright 2016-2019 JetBrains s.r.o.
- * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
- */
-
-package androidx.compose.runtime.external.kotlinx.collections.immutable.internal
-
-internal fun assert(@Suppress("UNUSED_PARAMETER") condition: Boolean) {}
```
Comment (eymar) [inline compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/external/kotlinx/collections/immutable/internal/commonFunctions.kt:8]: This function is removed here, but not added in another package. Is it not needed at all?

File: compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/external/kotlinx/collections/immutable/internal/commonFunctions.kt (hunk 1 above)
Comment (dima-avdeev-jb) [reply compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/external/kotlinx/collections/immutable/internal/commonFunctions.kt]: Sorry, on JVM all work's fine.
But if I simply add an import, JVM breaks.
So, I copied the expect function right in this file.

File: compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/external/kotlinx/collections/immutable/internal/commonFunctions.kt (hunk 2 above)
Comment (dima-avdeev-jb) [reply compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/external/kotlinx/collections/immutable/internal/commonFunctions.kt:8]: This function is copied to a single file where it is used

File: compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/external/kotlinx/collections/immutable/internal/commonFunctions.kt (hunk 1 above)
Comment (dima-avdeev-jb) [reply compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/external/kotlinx/collections/immutable/internal/commonFunctions.kt]: Error on Native: `Unresolved reference: modCount` on usage lines
`internal fun getModCount() = modCount`
`...and other usages...`

Comment (dima-avdeev-jb) [reply compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/external/kotlinx/collections/immutable/internal/commonFunctions.kt]: If I add import, then Native compiles fine.
But on JVM error: `Unresolved reference: modCount` on import line
`import androidx.compose.runtime.external.kotlinx.collections.immutable.internal.modCount`

Comment (igordmn) [reply compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/external/kotlinx/collections/immutable/internal/commonFunctions.kt]: I have a suspicion that it wil crash in runtime in Native/Web somewhere.

Let's add a task to check after we can build.

And if it crashes somewhere - fix somehow (return 0, for example)

File: compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/external/kotlinx/collections/immutable/internal/commonFunctions.kt (hunk 2 above)
Comment (dima-avdeev-jb) [reply compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/external/kotlinx/collections/immutable/internal/commonFunctions.kt:8]: Thanks — it is my mistake
Reverted back

**Review by eymar — APPROVED** [review]
I ran ./gradlew :compose:runtime:runtime:jsTest and they passed (after commenting a couple other jvm-only things)

File: compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/external/kotlinx/collections/immutable/internal/commonFunctions.kt (hunk 1 above)
Comment (igordmn) [reply compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/external/kotlinx/collections/immutable/internal/commonFunctions.kt]: https://youtrack.jetbrains.com/issue/COMPOSE-180/Check-if-native-target-crashes-when-use-modCount

**Review by igordmn — APPROVED** [review]

**Comment by dima-avdeev-jb** [comment]
I run an iOS Demo app and it works.
But it doesn't use this code in our Demo project.

**Pull request merged**